app/grandchallenge/core/error_handlers.py

```diff
@@ -0,0 +1,98 @@
+from grandchallenge.notifications.models import Notification, NotificationType
+
+
+class ErrorHandler:
+    def handle_error(self, *, error_message, user=None, interface=None):
+        raise NotImplementedError("Subclasses must implement this method")
+
+
+class JobCIVErrorHandler(ErrorHandler):
+
+    def __init__(self, job):
+        from grandchallenge.algorithms.models import Job
+
+        if not job or not isinstance(job, Job):
+            raise RuntimeError(
+                "You need to provide a Job instance to this error handler."
+            )
+
+        self._job = job
+
+    def handle_error(self, *, error_message, interface, user=None):
+        detailed_error_message = self._job.detailed_error_message.copy()

```

Comment:
`.copy()` is a shallow copy, you likely want to use `copy.deepcopy(self._job.detailed_error_message)`.